islandora_transforms/library/xslt-date-template.xslt

```diff
@@ -1,14 +1,101 @@
+    <xsl:variable name="pattern6">[0-9]{4}-[0-9]{2}/[0-9]{4}-[0-9]{2}</xsl:variable>
+    
+    <!-- A "catch all" for other probable date ranges and take the start of them -->
+    <xsl:variable name="pattern7">([0-9]{4})(-[0-9]{2})?(-[0-9]{2})?/.*</xsl:variable>
+
+    <!-- Have JODA or fail silently. -->
+
+    <xsl:variable name="parsed">
+      <xsl:choose>
+        <xsl:when test="java:matches($date, $pattern2)">
+          <xsl:variable name="dp"
+            select="java:org.joda.time.format.DateTimeFormat.forPattern('M/d/y')"/>
+          <xsl:value-of select="java:parseDateTime($dp, $date)"/>
+        </xsl:when>
+        <xsl:when test="java:matches($date, $pattern3)">
+          <xsl:variable name="dp"
+            select="java:org.joda.time.format.DateTimeFormat.forPattern('M/d/y H:m')"/>
+          <xsl:value-of select="java:parseDateTime($dp, $date)"/>
+        </xsl:when>
+        <xsl:when test="java:matches($date, $pattern4)">
+          <xsl:variable name="dp" select="java:org.joda.time.format.DateTimeFormat.forPattern('yyyy')"/>
+          <xsl:value-of select="java:parseDateTime($dp, substring-before($date, '/'))"/>
+        </xsl:when>
+        <xsl:when test="java:matches($date, $pattern5)">
+          <xsl:variable name="dp" select="java:org.joda.time.format.DateTimeFormat.forPattern('yyyy-MM-dd')"/>
+          <xsl:value-of select="java:parseDateTime($dp, substring-before($date, '/'))"/>
+        </xsl:when>
+        <xsl:when test="java:matches($date, $pattern6)">
+          <xsl:variable name="dp" select="java:org.joda.time.format.DateTimeFormat.forPattern('yyyy-MM')"/>
+          <xsl:value-of select="java:parseDateTime($dp, substring-before($date, '/'))"/>
+        </xsl:when>
+        <xsl:when test="java:matches($date, $pattern7)">
+          <xsl:value-of select="substring-before($date, '/')"/>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="$date"/>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:variable>
+    <xsl:variable name="asdf" select="string($parsed)"/>
+    <xsl:message>
+      <xsl:value-of select="$asdf"/>
+    </xsl:message>
+
+    <xsl:choose>
+      <xsl:when test="java:matches($asdf, $pattern1)">
+        <xsl:variable name="dp"
+          select="java:org.joda.time.format.ISODateTimeFormat.dateTimeParser()"/>
+
+        <!--<xsl:message><xsl:value-of select="java:parseDateTime($dp, $parsed)"/></xsl:message>-->
+        <xsl:variable name="f" select="java:org.joda.time.format.ISODateTimeFormat.dateTime()"/>
+        <xsl:variable name="df" select="java:withZoneUTC($f)"/>
+        <xsl:value-of select="java:print($df, java:parseDateTime($dp, $asdf))"/>
+      </xsl:when>
+      <xsl:otherwise>
+        <xsl:message>
+          <xsl:value-of select="$parsed"/>
+        </xsl:message>
+      </xsl:otherwise>
+    </xsl:choose>
 
-    <xsl:value-of select="java:ca.discoverygarden.gsearch_extensions.JodaAdapter.transformForSolr($date, $pid, $datastream)"/>
```

Comment:
@nigelgbanks: The only thing to note here is that this XSLT was used before Adam's JODA awesomeness moved out to a separate JAR which the date template here calls. This is something that could be an issue down the road if they want to be based off of our HEAD.